src/app/content/components/TableOfContents/keyboardSupport.hook.tsx

```diff
@@ -0,0 +1,171 @@
+import { HTMLElement, Element } from '@openstax/types/lib.dom';
+import { LinkedArchiveTreeNode } from '../../types';
+import React from 'react';
+
+export interface KeyboardSupportProps {
+  event: React.KeyboardEvent<HTMLAnchorElement>;
+  item: LinkedArchiveTreeNode;
+  isOpen?: boolean;
+  onSelect: () => void;
+}
+
+const getTreeItems = () => {
+  const treeItems = Array.from(document?.querySelectorAll('a[role="treeitem"]') as ArrayLike<HTMLElement>);
+  const treeItemResults = treeItems.filter((el) => {
+    const style = window?.getComputedStyle(el);
+    return (style?.display !== 'none');
```

Comment:
is there something you can add to the css selector like `[open]` to do this filter?

Reply:
You need to have different css selectors for the content links and the toggles, then you can do `a.content-link,a.toggle[open]` (or whatever the right selector would be

Reply:
Oh nevermind, I'm misinterpreting what you're trying to do. I think I see now you're trying to filter out the ones that are hidden.... I'll think about it some more